### DISCUSSION TITLE
Is there a way to call an API to external web https link and pass , receive the the status ?

### DISCUSSION THREAD

**Discussion by farookhus** [cloud-custodian/cloud-custodian#7372] [General]
I've deployed many custodian policies and these are getting executed via the lambda functions regularly. These lambda function get triggered based on the cloud trails or via corn schedules. I got a requirement to call an external web link API that will capture a value each time this lambda function is executed and this is mainly to capture the efforts and how many times these automation are being triggered. I'm looking for a help to find if there a existing ACTION available in the custodian to call the WEB API or we need to add additional scripts to achieve this requirement ? The external API authenticate with a custom credentials and we need to have options to pass the credentials while calling the API.

Appreciate your help of suggestion on this ?

**Comment by kapilt** [comment]
if your just looking to get a count of triggers, the cloud watch metrics on custodian already provide that. you can do a web hook action as well, see provider reference docs on the custodian website.

**Comment by farookhus** [comment]
> if your just looking to get a count of triggers, the cloud watch metrics on custodian already provide that. you can do a web hook action as well, see provider reference docs on the custodian website.

We have an centralized on-premises weblink to capture the other automation values as well , so I have to stick to the requirement of calling the weblink API to capture the count of triggers. I've used the webbook in cloud custodian method to post the data in the Microsoft team successfully but can you shoot me some examples code to call the webhook for weblink API along with the passing credentials.
